src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -142,22 +149,14 @@ public Task InitializeAsync()
             return Task.CompletedTask;
         }
 
-        public string GraphQLSchema()
+        private string GraphQLSchema()
         {
-            if (_cosmosDb.GraphQLSchema is null && _fileSystem.File.Exists(_cosmosDb.GraphQLSchemaPath))
-            {
-                _cosmosDb = _cosmosDb with { GraphQLSchema = _fileSystem.File.ReadAllText(_cosmosDb.GraphQLSchemaPath) };
-            }
-
-            if (_cosmosDb.GraphQLSchema is null)
+            if (_cosmosDb.GraphQLSchema is not null)
             {
-                throw new DataApiBuilderException(
-                    "GraphQL Schema isn't set.",
-                    System.Net.HttpStatusCode.InternalServerError,
-                    DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                return _cosmosDb.GraphQLSchema;
             }
 
```

Comment:
Is `GraphQLSchemaPath` replaced by the `Schema` property?

If the file exists, should we check if the contents are null, and error accordingly?